python/arcticdb/version_store/library.py

```diff
@@ -3021,3 +3026,28 @@ def _normalize_staged_data_mode(mode: Union[StagedDataFinalizeMethod, str]) -> S
             raise ArcticInvalidApiUsageException(
                 "mode must be one of StagedDataFinalizeMethod.WRITE, StagedDataFinalizeMethod.APPEND, 'write', 'append'"
             )
+
+    def merge(
```

Comment:
Should we have another method on this, vs making this a feature of `update`? 
Update already has an `upsert` parameter which many would think would have upsert semantics.

Reply:
Yeah, I find the semantics of the upsert parameter to be a bit confusing I guess the word got overloaded too much. Technically it's possible to add some flags and have this functionality as part of the update. I think the two things are doing quite different things and the implementation will be vastly different so I'm not quite convinced it'll be for the better to mash them into one API call.